core/io/image.cpp

```diff
@@ -3929,22 +3929,79 @@ void Image::renormalize_rgbe9995(uint32_t *p_rgb) {
 	// Never used
 }
 
+Image::FileFormat Image::detect_file_format(const uint8_t *p_data, int p_len) {
+	size_t len = p_len < 0 ? 0 : unsigned(p_len);
```

Comment:
Sometimes the image creation function is called without specifying the length of the data. In this case, the data length is set to -1. Therefore, I made length comparisons only if a positive data length is explicitly specified.